docs/dependency-injection-liftweb-scala.adoc

```diff
@@ -0,0 +1,382 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+= Dependency Injection in Lift
+
+== Introduction
+
+Dependency injection is an important topic in the JVM world. With Java, we usually go with Guice
+or Spring whenever we need it. However, scala has certain advantages over Java, like functions,
+that allow basic DI without resorting to entire DI frameworks.
+
+Scala allows the use of http://jonasboner.com/real-world-scala-dependency-injection-di/[cake pattern]
+that can provide DI by composing traits.
+
+Although the cake pattern is helpful, it doesn't provide complete DI functionality.
+The cake pattern allows you to compose complex classes out of Scala traits, but the it's less helpful
+in terms of allowing you to make dynamic choices about which combination of cake to vend in a given situation.
+Lift provides extra features that complete the dependency injection puzzle.
+
+== Why Lift DI
+
+Many people come to Scala from a Java background, so it's only natural to want to use Guice/Spring,
+which do a great job in the java world. And we can https://groups.google.com/forum/#!topic/liftweb/_lleL2xpCFU[use those],
+but there's a better, more direct way that's less magical.
+
+Lift's philosophy is to keep things as simple as possible (and no simpler), and allow full control
+to the developers over how things work. To paraphrase David Pollak,
+"The abstraction turtles have to end somewhere" and  lift's DI features end those turtles quite
+early.
+
+== Lift DI and various dependency scopes (e.g. from Spring)
+
+Lift's DI allows all the
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes[four most commonly used scopes] that
+are possible with spring:
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-singleton[singleton],
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-prototype[prototype], https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-request[request],
+and https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-session[session]. The others can be achieved
+relatively easily if you want.
```

Comment:
I would move this bit to the bottom, since it's more of a footnote IMO given the goal of introducing Lift's DI.